src/examples/libpmem2/map_multiple_files/map_multiple_files.c

```diff
@@ -0,0 +1,280 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * map_multiple_files.c -- implementation of virtual address allocation example
+ */
+
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#ifndef _WIN32
+#include <unistd.h>
+#else
+#include <io.h>
+#endif
+#include <libpmem2.h>
+
+/*
+ * cleanup_arrays - free dynamically allocated memory
+ */
+static void
+free_arrays(size_t **file_size, int **fd, struct pmem2_map ***map,
+		struct pmem2_source ***src)
+{
+	free(*file_size);
+	free(*fd);
+	free(*map);
+	free(*src);
+}
+
+/*
+ * init_arrays - initialize arrays
+ */
+static int
+init_arrays(size_t nfiles, size_t **file_size, int **fd,
+		struct pmem2_map ***map, struct pmem2_source ***src)
+{
+	int ret = 0;
+
+	*file_size = malloc(sizeof(size_t) * nfiles);
```

Comment:
consider if something like this would be more succinct:
```
struct file_dsc {
  int fd;
  size_t size;
  struct pmem2_source *src;
  struct pmem2_map *map;
};

int file_dsc_init(struct file_dsc *dsc, const char *path) { ... };
void file_dsc_fini(struct file_dsc *fdsc) { ... };

struct file_dsc *fdsc = ...;
int nfdsc = 0;
for n in 0...nfiles {
    if file_dsc_init(&fdsc[n], argv[n]) != 0 {
      goto fdsc_fini;
    } 
    nfdsc = n;
}

/* the actual example code */

fdsc_fini:
for n in nfdsc...0 {
    file_dsc_fini(&fdsc[n]);
}
```